src/devices/Bmxx80/Bme680.cs

```diff
@@ -102,38 +304,32 @@ public bool ReadHasNewData()
         /// <returns>Calculated humidity.</returns>
         public async Task<double> ReadHumidityAsync()
         {
+            if (HumiditySampling == Sampling.Skipped)
+                return double.NaN;
+
             // Read humidity data.
-            byte msb = Read8BitsFromRegister((byte)Bme680Register.HUMIDITYDATA_MSB);
-            byte lsb = Read8BitsFromRegister((byte)Bme680Register.HUMIDITYDATA_LSB);
+            var msb = Read8BitsFromRegister((byte)Bme680Register.HUMIDITYDATA_MSB);
```

Comment:
nit: many devices allow reading more bytes on the register address (so you pass address of whichever address is lower (MSB/LSB) and read 16bits and then you can use https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1 if needed.

I'm not sure if this applies here but might be worth checking